http/oidc/src/test/java/org/wildfly/security/http/oidc/OidcProviderMetadataTest.java

```diff
@@ -0,0 +1,374 @@
+        assertNull(emptyOidcProviderMetadata.getJwksUri());
+    }
+
+    @Test
+    public void testCheckSessionIframe() throws Exception {
+        assertTrue("http://localhost:8080/realms/myrealm/protocol/openid-connect/login-status-iframe.html".equals(oidcProviderMetadata.getCheckSessionIframe()));
+        assertTrue("http://localhost:8080/login-status-iframe.html".equals(withoutBooleansOidcProviderMetadata.getCheckSessionIframe()));
+        assertNull(emptyOidcProviderMetadata.getCheckSessionIframe());
+    }
+
+    @Test
+    public void testGrantTypesSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getGrantTypesSupported();
+        assertTrue(l.contains("refresh_token"));
+        assertNull(emptyOidcProviderMetadata.getGrantTypesSupported());
+    }
+
+    @Test
+    public void testResponseTypesSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getResponseTypesSupported();
+        assertTrue(l.contains("code id_token"));
+        assertNull(emptyOidcProviderMetadata.getResponseTypesSupported());
+    }
+
+    @Test
+    public void testSubjectTypesSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getSubjectTypesSupported();
+        assertTrue(l.contains("pairwise"));
+        assertNull(emptyOidcProviderMetadata.getSubjectTypesSupported());
+    }
+
+    @Test
+    public void testIdTokenSigningAlgValuesSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getIdTokenSigningAlgValuesSupported();
+        assertTrue(l.contains("HS256"));
+        assertNull(emptyOidcProviderMetadata.getIdTokenSigningAlgValuesSupported());
+    }
+
+    @Test
+    public void testIdTokenEncryptionAlgValuesSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getIdTokenEncryptionAlgValuesSupported();
+        assertTrue(l.contains("RSA1_5"));
+        assertNull(emptyOidcProviderMetadata.getIdTokenEncryptionAlgValuesSupported());
+    }
+
+    @Test
+    public void testIdTokenEncryptionEncValuesSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getIdTokenEncryptionEncValuesSupported();
+        assertTrue(l.contains("A128CBC-HS256"));
+        assertNull(emptyOidcProviderMetadata.getIdTokenEncryptionEncValuesSupported());
+    }
+
+    @Test
+    public void testUserInfoSigningAlgValuesSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getUserInfoSigningAlgValuesSupported();
+        assertTrue(l.contains("EdDSA"));
+        assertNull(emptyOidcProviderMetadata.getUserInfoSigningAlgValuesSupported());
+    }
+
+    @Test
+    public void testRequestObjectSigningAlgValuesSupported() throws Exception {
+        List<String> l = oidcProviderMetadata.getRequestObjectSigningAlgValuesSupported();
+        assertTrue(l.contains("RS384"));
+        assertNull(emptyOidcProviderMetadata.getRequestObjectSigningAlgValuesSupported());
```

Comment:
I think this needs an assertNull for withoutBooleansProviderMetaData.

Reply:
done